libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/TreeViewRenderer/TreeViewRenderer.tsx

```diff
@@ -91,6 +93,10 @@ export class TreeViewRenderer extends React.PureComponent<
   ITreeViewRendererProps,
   ITreeViewRendererState
 > {
+  public static contextType = ModelAssessmentContext;
```

Comment:
why do we have model assessment context stuff in TreeViewRenderer?  Just seems like a really bad composition of things.  This should be passed in through the props.

Reply:
The purpose of context is to avoid multi level prop pass to make code simpler.
Our design is to keep top level props from sdk to context so that whenever any component needs to reference we don't have to pass props though multi layer.
ModelAssesmentContext is just a naming, it is used in interpret and error analysis dashboard already.

Reply:
"The purpose of context is to avoid multi level prop pass to make code simpler."
sure, it makes it simpler, but it also makes it a lot more magical, harder to follow the code, and much less reusable.  To me it just seems extremely unexpected to see anything related to model assessment here.  If someone wanted to, they should be able to just take this component and reuse it in their code with the right properties passed in.  By having this context we are really making everything a lot more difficult from reusability point of view and encapsulation.